flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/utils/FlinkRexUtil.scala

```diff
@@ -535,6 +534,29 @@ object FlinkRexUtil {
         throw new IllegalArgumentException(s"Unknown expression type '${expr.getClass}': $expr")
     }
   }
+
+  /**
+   * Similar to [[RexUtil#findOperatorCall(SqlOperator, RexNode)]],
+   * but with a broader predicate support and returning a boolean.
+   */
+  def hasOperatorCallMatching(expr: RexNode, predicate: Predicate[SqlOperator]): Boolean = {
```

Comment:
Because we don't use scala-java8-compat, I did this, let me know if you have alternatives